src/graph/graph.c

```diff
@@ -310,17 +363,31 @@ int Graph_GetNodeLabel(const Graph *g, NodeID nodeID) {
 int Graph_GetEdgeRelation(const Graph *g, Edge *e) {
     assert(g && e);
     NodeID srcNodeID = Edge_GetSrcNodeID(e);
-    NodeID destNodeID = Edge_GetDestNodeID(e);
+    NodeID destNodeID = Edge_GetDestNodeID(e);    
+    EdgeID id = ENTITY_GET_ID(e);
 
     // Search for relation mapping matrix M, where
     // M[dest,src] == edge ID.
     for(int i = 0; i < array_len(g->_relations_map); i++) {
         EdgeID edgeId = 0;
         GrB_Matrix M = _Graph_GetRelationMap(g, i);
         GrB_Info res = GrB_Matrix_extractElement_UINT64(&edgeId, M, destNodeID, srcNodeID);
-        if(res == GrB_SUCCESS && edgeId == ENTITY_GET_ID(e)) {
+        if(res != GrB_SUCCESS) continue;
+
+        if(SINGLE_EDGE(edgeId) && (SINGLE_EDGE_ID(edgeId)) == id) {
```

Comment:
The second condition should be a separate `if` statement nested into the first here:
```
if(SINGLE_EDGE(edgeId)) {
    if ((SINGLE_EDGE_ID(edgeId)) == id) {
        Edge_SetRelationID(e, i);
        return i;
    }
} else {
```
Otherwise, we'll try to interpret single edges that fail the second condition as edge arrays.

Reply:
👍  good catch!